pd-client/leader_change_test.go

```diff
@@ -14,42 +14,113 @@
 package pd
 
 import (
-	"strings"
+	"os"
 	"time"
 
+	"github.com/coreos/etcd/clientv3"
 	. "github.com/pingcap/check"
+	"github.com/pingcap/pd/server"
 )
 
 var _ = Suite(&testLeaderChangeSuite{})
 
 type testLeaderChangeSuite struct{}
 
 func (s *testLeaderChangeSuite) TestLeaderChange(c *C) {
-	srv1 := newServer(c, 1235, "/pd-leader-change", 1)
+	etcdCfgs := server.NewTestMultiEtcdConfig(3)
 
-	// wait for srv1 to become leader
-	time.Sleep(time.Second * 3)
+	ch := make(chan *server.Server, 3)
+	rootPath := "/pd-leader-change"
 
-	client, err := NewClient(strings.Split(*testEtcd, ","), "/pd-leader-change", 1)
+	dirs := make([]string, 0, 3)
+	for i := 0; i < 3; i++ {
+		cfg := &server.Config{
+			Addr:            "127.0.0.1:0",
+			RootPath:        rootPath,
+			LeaderLease:     1,
+			TsoSaveInterval: 500,
+			ClusterID:       0,
+			EtcdCfg:         etcdCfgs[i],
+		}
+
+		dirs = append(dirs, cfg.EtcdCfg.DataDir)
+
+		go func() {
```

Comment:
we must start etcd concurrently, if not, starting etcd will block, because etcd will send message to other peer for starting.